src/InstrumentationEngine/Instruction.cpp

```diff
@@ -522,17 +522,18 @@ HRESULT MicrosoftInstrumentationEngine::CInstruction::GetIsRemoved(_Out_ BOOL* p
     return S_OK;
 }
 
-HRESULT MicrosoftInstrumentationEngine::CInstruction::GetInstructionSize(_Out_ DWORD* pdwSize)
+HRESULT MicrosoftInstrumentationEngine::CInstruction::GetInstructionSize(_In_ IInstruction* pInstruction, _Out_ DWORD* pdwSize)
```

Comment:
>GetInstructionSize(_In_ IInstruction* pInstruction, _Out_ DWORD* pdwSize) [](start = 54, length = 73)

I don't think that I understand the need for this. It is adding an additional if check for no reason.

Reply:
While refactoring ```CExceptionClause::RenderExceptionClause()``` to only operate on ```IInstruction``` I noticed the only required method from ```CInstruction``` that's not part of ```IInstruction``` interface was ```CInstruction::GetInstructionSize``` so I added this public static method.

Reply:
GetInstructionSize is called a lot in tight loops when decoding methods. Adding any additional overhead to it might cause slow down. I'm OK with this, but I might have to change it in my perf updates if it takes too long.

---
In reply to: [520888584](https://github.com/microsoft/CLRInstrumentationEngine/pull/325#discussion_r520888584) [](ancestors = 520888584)

Reply:
The compiler should be able to inline the indirection, but if it has a perf impact we'll figure out another solution